service/history/workflow/mutable_state_impl.go

```diff
@@ -6742,18 +6742,7 @@ func (ms *MutableStateImpl) disablingTransitionHistory() bool {
 //     completion. Exception: if Deployment is set but the workflow's effective behavior is
 //     UNSPECIFIED, it means the workflow is unversioned, so effective deployment will be nil.
 func (ms *MutableStateImpl) GetEffectiveDeployment() *deploymentpb.Deployment {
-	versioningInfo := ms.GetExecutionInfo().GetVersioningInfo()
-	if versioningInfo == nil {
-		return nil
-	} else if transition := versioningInfo.GetDeploymentTransition(); transition != nil {
-		return transition.GetDeployment()
-	} else if override := versioningInfo.GetVersioningOverride(); override != nil &&
-		override.GetBehavior() == enumspb.VERSIONING_BEHAVIOR_PINNED {
-		return override.GetDeployment()
-	} else if ms.GetEffectiveVersioningBehavior() != enumspb.VERSIONING_BEHAVIOR_UNSPECIFIED {
-		return versioningInfo.GetDeployment()
-	}
-	return nil
+	return GetEffectiveDeployment(ms.GetExecutionInfo().GetVersioningInfo())
```

Comment:
I don't like these names but I can't come up with something better.

Reply:
We can change later if you come up with something better :)